MimeKit/Cryptography/DefaultSecureMimeContext.cs

```diff
@@ -432,20 +432,19 @@ protected override ISet<TrustAnchor> GetTrustedAnchors ()
 		/// <returns>The intermediate certificates.</returns>
 		protected override IStore<X509Certificate> GetIntermediateCertificates ()
 		{
-			//var intermediates = new X509CertificateStore ();
-			//var selector = new X509CertStoreSelector ();
-			//var keyUsage = new bool[9];
+			var intermediates = new X509CertificateStore ();
+			var selector = new X509CertStoreSelector ();
+			var keyUsage = new bool[9];
 
-			//keyUsage[(int) X509KeyUsageBits.KeyCertSign] = true;
-			//selector.KeyUsage = keyUsage;
+			keyUsage[(int) X509KeyUsageBits.KeyCertSign] = true;
+			selector.KeyUsage = keyUsage;
 
-			//foreach (var record in dbase.Find (selector, false, X509CertificateRecordFields.Certificate)) {
-			//	if (!record.Certificate.IsSelfSigned ())
-			//		intermediates.Add (record.Certificate);
-			//}
+			foreach (var record in dbase.Find (selector, false, X509CertificateRecordFields.Certificate)) {
+				if (!record.Certificate.IsSelfSigned ())
+					intermediates.Add (record.Certificate);
+			}
 
-			//return intermediates;
-			return dbase;
+			return intermediates;
 		}
```

Comment:
Hard to argue with the logic I used to have there (but for some reason commented out). Did you look to see why that code was commented out? I'm guessing it had something to do with it not getting the results that I was expecting it to get.

Hmmm, looks like it was this commit. I'm not always great at documenting my changes :(

> commit ac9dce130be10fca2257f82b6ed1910a358c1c9d
> Date:   Tue Nov 5 10:37:22 2019 -0500
> 
>    Go back to using X509CertStoreSelector.Certificate when building a cert chain

Looks like before that, I had been trying to use a selector with:

```csharp
selector.Subject = certificate.SubjectDN;
```

I guess I'll assume that my original logic was fine and that un-commenting this won't actually break anything.